c7n/resources/asg.py

```diff
@@ -492,6 +492,53 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('launch-failure')
+class LaunchFailureFilter(Filter):
+    """
+    This filter is designed to report ASGs that are constantly trying to
+    spin up new instances & failing to do so over & over.
+    The filter can be used to search for failure events over a period of time
+    by using the 'days' parameter.
+
+    - name: asg-launch-failure
+      resource: asg
+      filters:
+        - launch-failure
+          days: 10
+    """
+    schema = type_schema(
+        'launch-failure',
+        days={'type': 'number'})
+
+    def asg_activities(self, asg):
+        client = local_session(
+            self.manager.session_factory).client('autoscaling')
+        p = client.get_paginator('describe_scaling_activities')
+        results = {}
+        for activity in p.paginate(
+                AutoScalingGroupName=asg['AutoScalingGroupName']):
+            for a in activity['Activities']:
+                if a['StartTime'].replace(tzinfo=tzutc()) < datetime.now(
+                        tz=tzutc()) - timedelta(days=self.data.get('days', 1)):
+                    break
+                if 'Launching a new EC2 instance' not in a['Description']:
+                    continue
+                if a['StatusCode'] != 'Failed':
+                    continue
+                results.setdefault(a['AutoScalingGroupName'], []).append(
+                    a['ActivityId'])
+        return {k:v for k, v in results.items()}
+
+    def process(self, asgs, event=None):
+        with self.executor_factory(max_workers=3) as w:
+            failures = w.map(self.asg_activities, asgs)
```

Comment:
with the limits that are availabe for asgs, what would you recommend setting the chunk size to?